index.bs

```diff
@@ -1450,6 +1451,60 @@ that are returned to the caller when a new credential is created, or a new asser
         [=extension identifier=] → [=client extension output=] entries produced by the extension's
         [=client extension processing=].
 
+    :   {{PublicKeyCredential/toJSON()}}
+    ::  This operation returns {{RegistrationResponseJSON}} or {{AuthenticationResponseJSON}}, that is a JSON-serialized
+        representation of the instance of {{PublicKeyCredential}} suitable for submission to a [=[RP]=] server as a DOMString
+        payload with `application/json` mimetype. The [=client=] is in charge of
+        [serializing values to JSON as usual](https://webidl.spec.whatwg.org/#idl-tojson-operation), but must take additional
+        steps to encode any {{ArrayBuffer}} values to {{DOMString}} values using [=base64url encoding=] to ensure compatibility
```

Comment:
I assume we don't *actually* want Base64URL here but are using it for consistency only. I.e. We're not expecting developers want to put these json strings into URL parameters, right?

Would it be worth to use regular Base64 in order to simplify use of standard server-side JSON parsers? Go's encoding/json for example automatically encodes byte slices into base64 (not base64url) strings ([see docs here](https://pkg.go.dev/encoding/json#:~:text=Array%20and%20slice%20values%20encode%20as%20JSON%20arrays%2C%20except%20that%20%5B%5Dbyte%20encodes%20as%20a%20base64%2Dencoded%20string%2C%20and%20a%20nil%20slice%20encodes%20as%20the%20null%20JSON%20value.). So going with Base64url could potentially be a minor annoyance for server developers (extra code to write; potential source for subtle bugs).

The obvious drawback is that the spec already uses Base64URL elsewhere, so mixing the two could be mildly confusing. Though we could call out the difference and explain the reasoning behind it if we wanted.

Reply:
> The obvious drawback is that the spec already uses Base64URL elsewhere, so mixing the two could be mildly confusing.

This is exactly why I've chosen to use Base64URL across multiple implementations despite Base64 being workable without external dependencies or custom code, because [the spec lists it as a dependency](https://www.w3.org/TR/webauthn-2/#sctn-dependencies).

> We're not expecting developers want to put these json strings into URL parameters, right?

Definitely not, I'm almost certain the output would be conveyed in the body of a POST request.

> Though we could call out the difference and explain the reasoning behind it if we wanted.

I think it's asking for a bad time if we mix-and-match encodings, and the ship has probably sailed for switching WebAuthn to use Base64 instead (I agree with you that more languages support Base64 out of the box and probably would have advocated for Base64 had I been fortunate to be involved in those early days.) I'd argue requiring RP's to keep track of when to use Base64 vs Base64URL to encoding or decoding would push this feature the other way, towards making WebAuthn more challenging to use.